cmd/iam-keys-check/main.go

```diff
@@ -0,0 +1,230 @@
+	Lambda          bool   `short:"l" long:"lambda" description:"Run as an AWS lambda function." required:"false" env:"LAMBDA"`
+	MaxDays         uint   `long:"days" description:"The maximum age in days that a key can be active without triggering an alert." default:"90" env:"MAX_DAYS"`
+	PollInterval    uint   `long:"poll-interval" description:"The poll interval in milliseconds when checking if a credential report is available." default:"5000" env:"POLL_INTERVAL"`
+	SlackWebhookURL string `long:"slack-webhook-url" description:"The Slack Webhook Url." required:"true" env:"SLACK_WEBHOOK_URL"`
+	SlackChannel    string `long:"slack-channel" description:"The Slack channel." required:"true" env:"SLACK_CHANNEL"`
+}
+
+var options Options
+var logger *zap.Logger
+
+func rowToMap(header []string, row []string) map[string]string {
+	m := map[string]string{}
+	for i, h := range header {
+		m[strings.ToLower(h)] = row[i]
+	}
+	return m
+}
+
+func getCredentialReport(iamClient *iam.IAM, tries int, pollInterval int) (*iam.GetCredentialReportOutput, error) {
+	if tries <= 0 {
+		return &iam.GetCredentialReportOutput{}, errors.New("maxmimum number of tries to get credential report reached")
+	}
+	report, err := iamClient.GetCredentialReport(&iam.GetCredentialReportInput{})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == iam.ErrCodeCredentialReportNotPresentException || aerr.Code() == iam.ErrCodeCredentialReportExpiredException {
+				_, err = iamClient.GenerateCredentialReport(&iam.GenerateCredentialReportInput{})
+				if err != nil {
+					if aerr, ok = err.(awserr.Error); ok {
+						if aerr.Code() == iam.ErrCodeLimitExceededException {
+							time.Sleep(time.Duration(pollInterval) * time.Millisecond)
+							return getCredentialReport(iamClient, tries-1, pollInterval)
+						}
+					}
+					log.Fatal(err)
+				}
+			} else if aerr.Code() == iam.ErrCodeCredentialReportNotReadyException {
+				time.Sleep(time.Duration(pollInterval) * time.Millisecond)
+				return getCredentialReport(iamClient, tries-1, pollInterval)
+			}
+		}
+		log.Fatal(err)
+	}
+	return report, nil
+}
+
+func sendAlertToSlack(usersSet stringSet, maxDays float64) error {
+
+	payload := map[string]interface{}{
+		"channel": options.SlackChannel,
+		"text":    "AWS notification",
+		"attachments": []map[string]string{
+			map[string]string{
+				"title": "Message",
+				"text":  "The following users have an active access key over " + fmt.Sprint(int(maxDays)) + " days old: " + strings.Join(usersSet.toArray(true), ", ") + "",
+			},
+		},
+	}
+
+	payloadJSON, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+
+	_, err = http.Post(options.SlackWebhookURL, "application/json", bytes.NewBuffer(payloadJSON))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func triggerCheck() {
+
+	maxDays := float64(options.MaxDays)
+
+	if maxDays == 0 {
```

Comment:
Still not being used as MaxDays is a `uint`

Reply:
?  An unsigned int can be zero.

Reply:
err sorry, I misread the change.